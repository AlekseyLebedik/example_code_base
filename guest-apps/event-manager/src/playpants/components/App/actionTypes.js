const prefix = 'playpants/APP';
export const FETCH_PROJECT_SETTINGS = `${prefix}.PROJECT_SETTINGS`;
export const FETCH_TEMPLATES = `${prefix}.TEMPLATES`;
export const FETCH_STORIES = `${prefix}.STORIES`;
export const FETCH_SCHEDULES = `${prefix}.SCHEDULES`;
export const UPDATE_SCHEDULES = `${prefix}.UPDATE_SCHEDULES`;
