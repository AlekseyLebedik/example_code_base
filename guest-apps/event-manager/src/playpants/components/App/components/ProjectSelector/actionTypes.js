export const CHANGE_PROJECT = 'PROJECT_SELECTOR.CHANGE_PROJECT';
export const CONFIGURED_PROJECTS_FETCH = 'CONFIGURED_PROJECTS_FETCH';
export const CONFIGURED_PROJECTS_FETCH_SUCCESS =
  'CONFIGURED_PROJECTS_FETCH_SUCCESS';
export const CONFIGURED_PROJECTS_FETCH_FAILED =
  'CONFIGURED_PROJECTS_FETCH_FAILED';
