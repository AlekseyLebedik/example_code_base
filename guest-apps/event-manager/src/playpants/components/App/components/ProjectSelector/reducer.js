import {
  CHANGE_PROJECT,
  CONFIGURED_PROJECTS_FETCH_SUCCESS,
} from './actionTypes';

const initialState = {
  contentTypeId: null,
  companies: [],
  id: null,
  name: null,
  titles: [],
  configuredProjects: [],
};

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case CHANGE_PROJECT:
      return {
        ...state,
        ...action.project,
      };
    case CONFIGURED_PROJECTS_FETCH_SUCCESS:
      return {
        ...state,
        configuredProjects: [...state.configuredProjects, ...action.projects],
      };
    default:
      return state;
  }
}
