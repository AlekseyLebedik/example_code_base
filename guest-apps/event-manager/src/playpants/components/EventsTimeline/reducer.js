import { combineReducers } from 'redux';
import conflictsReducer from './conflictsSlice';

const reducer = combineReducers({
  conflicts: conflictsReducer,
});

export default reducer;
