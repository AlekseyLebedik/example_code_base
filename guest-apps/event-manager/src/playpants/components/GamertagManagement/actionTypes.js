const prefix = 'playpants/GAMERTAG_MANAGEMENT';

export const FETCH_GROUPS = `${prefix}.GROUPS`;
export const CREATE_GROUP = `${prefix}.CREATE_GROUP`;
export const DELETE_GROUP = `${prefix}.DELETE_GROUP`;
export const SEARCH_GROUPS = `${prefix}.SEARCH_GROUPS`;
export const CLEAR_GROUPS = `${prefix}.CLEAR_GROUPS`;

export const FETCH_GROUP_ACCOUNTS = `${prefix}.GROUP_ACCOUNTS`;
export const UPDATE_GROUP_ACCOUNTS = `${prefix}.UPDATE_GROUP_ACCOUNTS`;
export const CLEAR_GROUP_ACCOUNTS = `${prefix}.CLEAR_GROUP_ACCOUNTS`;

export const FETCH_GROUP_TIMEWARP_SETTINGS = `${prefix}.GROUP_TIMEWARP_SETTINGS`;
export const UPDATE_GROUP_TIMEWARP_SETTINGS = `${prefix}.UPDATE_GROUP_TIMEWARP_SETTINGS`;
export const CLEAR_GROUP_TIMEWARP_SETTINGS = `${prefix}.CLEAR_GROUP_TIMEWARP_SETTINGS`;
