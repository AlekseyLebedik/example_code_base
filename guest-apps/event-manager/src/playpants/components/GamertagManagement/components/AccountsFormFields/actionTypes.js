const prefix = 'playpants/GAMERTAGS_ACCOUNTS_FORM';

export const ASYNC_FETCH_LINKED_ACCOUNTS_LIST = `${prefix}.ASYNC_LINKED_ACCOUNTS_LIST_FETCH`;
export const ASYNC_FETCH_LINKED_ACCOUNTS_LIST_SUCCESS = `${prefix}.ASYNC_LINKED_ACCOUNTS_LIST_FETCH_SUCCESS`;
export const ASYNC_FETCH_LINKED_ACCOUNTS_LIST_FAILED = `${prefix}.ASYNC_LINKED_ACCOUNTS_LIST_FETCH_FAILED`;

export const ASYNC_FETCH_ACCOUNTS_GROUP = `${prefix}.ASYNC_ACCOUNTS_GROUP_FETCH`;
export const ASYNC_FETCH_ACCOUNTS_GROUP_SUCCESS = `${prefix}.ASYNC_ACCOUNTS_GROUP_FETCH_SUCCESS`;
export const ASYNC_FETCH_ACCOUNTS_GROUP_FAILED = `${prefix}.ASYNC_ACCOUNTS_GROUP_FETCH_FAILED`;
