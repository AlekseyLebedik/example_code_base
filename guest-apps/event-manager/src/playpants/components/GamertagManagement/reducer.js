import reduceReducers from 'reduce-reducers';
import { combineReducers } from 'redux';
import {
  createFetchReducer,
  createUpdateReducer,
} from '@demonware/devzone-core/helpers/reducers';

import * as AT from './actionTypes';

const groupsReducer = createFetchReducer(AT.FETCH_GROUPS);
const clearGroupsReducer = createFetchReducer(AT.CLEAR_GROUPS);

const accountsReducer = createFetchReducer(AT.FETCH_GROUP_ACCOUNTS);
const updateAccountsReducer = createUpdateReducer(AT.UPDATE_GROUP_ACCOUNTS);
const clearAccountsReducer = createFetchReducer(AT.CLEAR_GROUP_ACCOUNTS);

const timewarpSettingsReducer = createFetchReducer(
  AT.FETCH_GROUP_TIMEWARP_SETTINGS
);
const updatetimewarpSettingsReducer = createUpdateReducer(
  AT.UPDATE_GROUP_TIMEWARP_SETTINGS
);
const cleartimewarpSettingsReducer = createFetchReducer(
  AT.CLEAR_GROUP_TIMEWARP_SETTINGS
);

export default combineReducers({
  accounts: reduceReducers(
    accountsReducer,
    updateAccountsReducer,
    clearAccountsReducer
  ),
  groups: reduceReducers(groupsReducer, clearGroupsReducer),
  timewarpSettings: reduceReducers(
    timewarpSettingsReducer,
    updatetimewarpSettingsReducer,
    cleartimewarpSettingsReducer
  ),
});
