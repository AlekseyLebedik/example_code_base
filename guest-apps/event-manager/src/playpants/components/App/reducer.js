import isEmpty from 'lodash/isEmpty';
import { combineReducers } from 'redux';
import reduceReducers from 'reduce-reducers';
import {
  createFetchReducer,
  INITIAL_STATE,
} from '@demonware/devzone-core/helpers/reducers';
import { UPDATE_PROJECT_SETTING_SUCCEED } from 'playpants/scenes/ProjectSettings/actionTypes';

import { DEFAULT_PROJECT_SETTINGS } from 'playpants/constants/projectSettings';
import { CHANGE_PROJECT } from './components/ProjectSelector/actionTypes';
import projectSelectorReducer from './components/ProjectSelector/reducer';

import * as AT from './actionTypes';
import { parseProjectSettings, parseProjectSettingsHelper } from './helpers';

const templatesReducer = createFetchReducer(AT.FETCH_TEMPLATES);
const projectSettingsReducer = createFetchReducer(AT.FETCH_PROJECT_SETTINGS);
const storiesReducer = createFetchReducer(AT.FETCH_STORIES);
const schedulesReducer = createFetchReducer(AT.FETCH_SCHEDULES);

const PROJECT_SETTINGS_INITIAL_STATE = {
  ...INITIAL_STATE,
  data: [DEFAULT_PROJECT_SETTINGS],
};

const AppReducer = (state = PROJECT_SETTINGS_INITIAL_STATE, action) => {
  switch (action.type) {
    case `${AT.FETCH_PROJECT_SETTINGS}_FETCH`:
    case `${AT.FETCH_PROJECT_SETTINGS}_FETCH_SUCCESS`:
    case `${AT.FETCH_PROJECT_SETTINGS}_FETCH_FAILED`:
      return {
        ...state,
        ...projectSettingsReducer(state, action),
        data: !isEmpty(action.data)
          ? parseProjectSettings(action.data)
          : PROJECT_SETTINGS_INITIAL_STATE.data,
      };
    case UPDATE_PROJECT_SETTING_SUCCEED: {
      const { data } = action.settings;
      return {
        ...state,
        data: state.data.map(setting =>
          setting.id === data.id ? parseProjectSettingsHelper(data) : setting
        ),
      };
    }
    case CHANGE_PROJECT:
      return {
        ...PROJECT_SETTINGS_INITIAL_STATE,
      };
    default:
      return {
        ...state,
      };
  }
};

const updateSchedulesReducer = (state = INITIAL_STATE, action) => {
  switch (action.type) {
    case AT.UPDATE_SCHEDULES:
      return {
        ...state,
        data: [...state.data, action.schedule],
      };
    default:
      return state;
  }
};

const reducer = combineReducers({
  currentProject: projectSelectorReducer,
  templates: templatesReducer,
  stories: storiesReducer,
  schedules: reduceReducers(schedulesReducer, updateSchedulesReducer),
  projectSettings: AppReducer,
});

export default reducer;
