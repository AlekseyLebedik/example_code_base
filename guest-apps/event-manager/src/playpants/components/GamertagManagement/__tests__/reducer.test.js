import mockState from 'playpants/testUtils/mockState';
import reducer from '../reducer';
import * as AT from '../actionTypes';

const { GamertagManagement } = mockState.Scenes.ProjectSettings;

describe('Groups', () => {
  describe('groupsReducer', () => {
    const initialState = {
      ...reducer(undefined, {}),
      ...GamertagManagement,
    };

    it('returns default state', () => {
      const state = reducer(undefined, {});
      expect(state).toMatchSnapshot();
    });

    describe('FETCH_GROUPS', () => {
      it('initiates fetch for groups', () => {
        const action = {
          type: `${AT.FETCH_GROUPS}_FETCH`,
          urlID: null,
          params: {},
          append: false,
        };
        const state = reducer(initialState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('FETCH_GROUPS_SUCCESS', () => {
      it('sets the groups with data', () => {
        const action = {
          type: `${AT.FETCH_GROUPS}_FETCH_SUCCESS`,
          data: [],
          next: null,
          append: false,
        };
        const state = reducer(initialState, action);
        expect(state).toMatchSnapshot();
      });
    });

    describe('FETCH_GROUPS_FAILED', () => {
      it('dispatches fetch error', () => {
        const action = {
          type: `${AT.FETCH_GROUPS}_FETCH_FAILED`,
        };
        const state = reducer(initialState, action);
        expect(state).toMatchSnapshot();
      });
    });
  });
});
