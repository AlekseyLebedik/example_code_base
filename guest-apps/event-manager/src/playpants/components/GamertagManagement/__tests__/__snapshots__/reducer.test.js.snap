// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Groups groupsReducer FETCH_GROUPS initiates fetch for groups 1`] = `
Object {
  "accounts": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "groups": Object {
    "data": Array [],
    "error": false,
    "loading": true,
    "next": undefined,
    "nextPageToken": undefined,
    "params": Object {},
  },
  "timewarpSettings": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
}
`;

exports[`Groups groupsReducer FETCH_GROUPS_FAILED dispatches fetch error 1`] = `
Object {
  "accounts": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "groups": Object {
    "data": Array [],
    "error": true,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "timewarpSettings": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
}
`;

exports[`Groups groupsReducer FETCH_GROUPS_SUCCESS sets the groups with data 1`] = `
Object {
  "accounts": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "groups": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": null,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "timewarpSettings": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
}
`;

exports[`Groups groupsReducer returns default state 1`] = `
Object {
  "accounts": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "groups": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
  "timewarpSettings": Object {
    "data": Array [],
    "error": false,
    "loading": false,
    "next": undefined,
    "nextPageToken": undefined,
    "params": undefined,
  },
}
`;
